local/node_process.go

```diff
@@ -90,30 +95,46 @@ type nodeProcess struct {
 	closedOnStop chan struct{}
 }
 
-func newNodeProcess(name string, log logging.Logger, cmd *exec.Cmd) (*nodeProcess, error) {
+func newNodeProcess(
+	name string,
+	log logging.Logger,
+	cmd *exec.Cmd,
+	startupTime time.Duration,
+) (*nodeProcess, error) {
 	np := &nodeProcess{
 		name:         name,
 		log:          log,
 		cmd:          cmd,
 		closedOnStop: make(chan struct{}),
 	}
-	return np, np.start()
+	return np, np.start(startupTime)
 }
 
 // Start this process.
 // Must only be called once.
-func (p *nodeProcess) start() error {
+func (p *nodeProcess) start(
+	startupTime time.Duration,
+) error {
 	p.lock.Lock()
-	defer p.lock.Unlock()
 
 	p.state = status.Running
 	if err := p.cmd.Start(); err != nil {
 		p.state = status.Stopped
 		close(p.closedOnStop)
+		p.lock.Unlock()
 		return fmt.Errorf("couldn't start process: %w", err)
 	}
 
 	go p.awaitExit()
+	p.lock.Unlock()
+	time.Sleep(startupTime)
```

Comment:
not sure If I like this approach.
to force it to  sleep for period of time.

we can have a go routine to check on the background if process is started and exit, but not sleep imho

pls feel free to ignore this comment if I didn't read it correctly

Reply:
I want this function to return on error if the process stopped before the given startup time.
So the function is forced to wait that time in the happy path. The only thing I can do with a busy
wait loop is to finish early in the non happy path (the process fails before the startup time). But for that
case I am not so much interested in faster times. So it is a simplest implementation.

Reply:
Anyway I will reduce this wait to one second as that is the time needed for the failure I am interested in
(ports conflict). And I am adding a busy wait of longer time for the wait for the process context
file, that takes more time in some executions.